File: src/game/powerups/TransparentDestroyableBlocksPowerUp.java
package game.powerups;

import game.entity.blocks.DestroyableBlock;
import game.entity.models.BomberEntity;
import game.models.Coordinates;
import game.utils.Paths;

import java.awt.image.BufferedImage;

public class TransparentDestroyableBlocksPowerUp extends PowerUp {
    public TransparentDestroyableBlocksPowerUp(Coordinates coords){
        super(coords);
    }

    @Override
    public BufferedImage getImage() {
        return loadAndSetImage(Paths.getPowerUpsFolder() + "/blocks_up.gif");
    }

    @Override
    public int getDuration() {
        return DEFAULT_DURATION_SEC;
    }

    @Override
    protected void doApply(BomberEntity entity) {
        entity.getWhiteListObstacles().remove(DestroyableBlock.class);
    }

    @Override
    protected void cancel(BomberEntity entity) {
        entity.getWhiteListObstacles().remove(DestroyableBlock.class);
    }
}

File: src/game/powerups/TransparentDestroyableBlocksPowerUp.kt
package game.powerups

import game.entity.EntityTypes
import game.entity.blocks.DestroyableBlock
import game.entity.player.BomberEntity
import game.entity.models.Coordinates
import game.utils.Paths.powerUpsFolder
import java.awt.image.BufferedImage

class TransparentDestroyableBlocksPowerUp : PowerUp {
    constructor(id: Long) : super(id)
    constructor(coordinates: Coordinates?) : super(coordinates)

    override fun getImage(): BufferedImage = loadAndSetImage("$powerUpsFolder/blocks_up.gif")

    override val duration: Int = 0

    override fun doApply(entity: BomberEntity) {
        entity.addWhiteListObstacle(DestroyableBlock::class.java)
    }

    override fun cancel(entity: BomberEntity) {
        entity.removeWhiteListObstacle(DestroyableBlock::class.java)
    }

    override val type: EntityTypes
        get() = EntityTypes.TransparentDestroyableBlocksPowerUp
}